### ISSUE TITLE
two networks

### ISSUE THREAD

**Issue by Yangjher** [HenryPengZou/JointMatch#4]
Hello, I have a question: If both networks, g and f, are initialized using bert-base-uncased, then they will produce the same output for the same input, meaning the backpropagation will also be the same.

**Comment by Yangjher** [comment]
The specific issue is located here.

![Image](https://github.com/user-attachments/assets/456a2777-34e9-40ec-a17c-4c098279545d)

**Comment by Yangjher** [comment]
![Image](https://github.com/user-attachments/assets/791daf3e-6e94-465d-aaac-b4f4d5efa7c6)

**Comment by HenryPengZou** [comment]
Hi @Yangjher, the initialization is the same for both networks, g and f, but the data used to train them is different (different pseudo-labeled data), thus their weights and gradients to be backpropagated are also different. 

Feel free to let me know if you have any additional questions or need help in understanding the code or paper~
